newrelic/hooks/external_httpx.py

```diff
@@ -12,23 +12,58 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from newrelic.common.object_wrapper import wrap_function_wrapper
 from newrelic.api.external_trace import ExternalTrace
+from newrelic.api.transaction import current_transaction
+from newrelic.common.object_wrapper import wrap_function_wrapper
 
 
 def bind_request(request, *args, **kwargs):
     return request
 
 
 def sync_send_wrapper(wrapped, instance, args, kwargs):
+    transaction = current_transaction()
     request = bind_request(*args, **kwargs)
-    with ExternalTrace('httpx', str(request.url), request.method):
+    connection = instance
+
+    with ExternalTrace("httpx", str(request.url), request.method) as tracer:
+        # Add the tracer to the connection object. The tracer will be
+        # used in getresponse() to add back into the external trace,
+        # after the trace has already completed, details from the
+        # response headers.
+        if hasattr(tracer, "generate_request_headers"):
+            outgoing_headers = dict(tracer.generate_request_headers(transaction))
+
+            # Preserve existing headers and add our outgoing headers
+            client_headers = request.headers
+            client_headers.update(outgoing_headers)
+            request.headers = client_headers
```

Comment:
No need to re-assign to `request.headers` here since it should be updated in-place.